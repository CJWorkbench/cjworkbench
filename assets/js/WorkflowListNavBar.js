import React from 'react'
import WfHamburgerMenu from './WfHamburgerMenu'

export default class WorkflowListNavBar extends React.Component {
  render() {
    // For now, we're hiding Lessons; so there are no links on the page. Only
    // show the links when we're on the /lessons/ page.
    const hideLinksBecauseFeatureIsNotFinished = window.location.href.indexOf('/lessons/') === -1

    const activeLink = window.location.href.indexOf('/lessons') === -1 ? 'workflows' : 'lessons'
    const propsForLink = (name) => {
      const activeProps = { href: null, className: 'active' }
      const inactiveProps = { className: 'inactive' }
      return activeLink === name ? activeProps : inactiveProps
    }
    const links = (
      <div className="links">
        <a href="/workflows/" {...propsForLink('workflows')} className="WF-toggle--link">
          <div>WORKFLOWS</div>
          <div className='WF-link--under'></div>
        </a>
        <a href="/lessons/" {...propsForLink('lessons')} className="LS-toggle--link">
          <div>TRAINING</div>
          <div className='LS-link--under'></div>
        </a>
      </div>
    )

    return (
      <div>
        <nav className="navbar">
          <div className="d-flex align-items-center">
            <img src="/static/images/logo.svg" className="logo"/>
            <div className="logo-1">Workbench</div>
          </div>
          {links}
          <WfHamburgerMenu />
        </nav>
      </div>
    )
  }
}
