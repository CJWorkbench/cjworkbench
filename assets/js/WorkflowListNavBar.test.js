import React from 'react'
import WorkflowListNavBar from './WorkflowListNavBar'
import { shallow } from 'enzyme'

describe('WorkflowListNavBar', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(<WorkflowListNavBar />)
    expect(wrapper).toMatchSnapshot()
  })
})
