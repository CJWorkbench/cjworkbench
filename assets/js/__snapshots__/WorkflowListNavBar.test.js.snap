// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowListNavBar Renders correctly 1`] = `
<div>
  <nav
    className="navbar"
  >
    <div
      className="d-flex align-items-center"
    >
      <img
        className="logo"
        src="/static/images/logo.svg"
      />
      <div
        className="logo-1"
      >
        Workbench
      </div>
    </div>
    <div
      className="links"
    >
      <div
        className="WF-toggle--link"
      >
        <a
          className="active"
          href={null}
        >
          WORKFLOWS
        </a>
        <div
          className="WF-link--under"
        />
      </div>
      <div
        className="LS-toggle--link"
      >
        <a
          className="inactive"
          href="/lessons/"
        >
          LESSONS
        </a>
        <div
          className="LS-link--under"
        />
      </div>
    </div>
    <WfHamburgerMenu />
  </nav>
</div>
`;
